pyk/src/pyk/proof/proof.py

```diff
@@ -287,6 +290,62 @@ def get_steps(self) -> Iterable[PS]:
         ...
 
 
+class MultiProof(Proof[None, None]):
+    """Thin concrete Proof class that has no execution logic of its own, but holds subproofs. The intended use
+    case for this is when we run kontrol proofs with setUp functions, to separate the proof into several
+    subproof APRProofs: one for the setUp function and one for the test function for each final configuration
+    of the setUp function.
```

Comment:
(Continuing the discussion here to form a thread.)

What feature of a `Proof` does this class rely on, what makes it different from, say, a `list[Proof]`?

Reply:
It would be the logic in `Proof.status` which tracks the status of the subproofs to tell when the whole proof has been discharged, and the ability to save/load from disk.